src/smt.cpp

```diff
@@ -119,8 +119,12 @@ vector<Expr> from1DIdx(
   // Start from the lowest dimension
   for (size_t ii = dims.size(); ii > 0; --ii) {
     size_t i = ii - 1;
-    idxs.emplace_back(i == 0 ? idx1d : idx1d.urem(dims[i]));
-    idx1d = idx1d.udiv(dims[i]);
+    uint64_t v;
+    auto isOne = dims[i].isUInt(v) && v == 1;
+
+    // how to make Expr(0)?
+    idxs.emplace_back(i == 0 ? idx1d : (isOne ? Expr::mkBV(0, 32U) : idx1d.urem(dims[i])));
```

Comment:
You can use `Expr::mkBV(0, dims[0])` to use `dims[0]`'s sort.
BTW, does this improve performance?

Reply:
printed expr and found (urem var #0001). isn't this 0?